sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetDataSourceV2.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources.v2.parquet
+
+import org.apache.spark.sql.execution.datasources._
+import org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat
+import org.apache.spark.sql.execution.datasources.v2._
+import org.apache.spark.sql.sources.v2.Table
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class ParquetDataSourceV2 extends FileDataSourceV2 {
+
+  override def fallbackFileFormat: Class[_ <: FileFormat] = classOf[ParquetFileFormat]
+
+  override def shortName(): String = "parquet"
+
+  override def getTable(options: CaseInsensitiveStringMap): Table = {
+    val paths = getPaths(options)
+    val tableName = getTableName(paths)
+    ParquetTable(tableName, sparkSession, options, paths, None, fallbackFileFormat)
+  }
+
+  override def getTable(options: CaseInsensitiveStringMap, schema: StructType): Table = {
+    val paths = getPaths(options)
+    val tableName = getTableName(paths)
+    ParquetTable(tableName, sparkSession, options, paths, Some(schema), fallbackFileFormat)
```

Comment:
What does this path do?

Parquet files always have their own schema and if I remember correctly, the Parquet read path doesn't support type coercion or files that are missing fields (maybe this was updated?). If that's the case, then shouldn't this path throw an exception?

Reply:
>  the Parquet read path doesn't support type coercion or files that are missing fields 
Yes, this is the current status.
But as a feature, I think we still need to handle the user-specified schema. The provided schema could be a subset of the inferred schema

Reply:
@gengliangwang, that is not how the DSv2 API works. The user-specified schema is not used to supply a projection. It is used to fill in missing schema details where Spark cannot infer them. This should be removed.

Reply:
> @gengliangwang, that is not how the DSv2 API works. The user-specified schema is not used to supply a projection. It is used to fill in missing schema details where Spark cannot infer them. This should be removed.

If I understand correctly, that is to say, we won't support the following for Parquet V2?
```
spark.read.schema(schema).parquet(...)
```

Reply:
That pattern is only supported in v1 under very strict requirements for the schema. It makes no sense to continue pretending that it is available in v2.

Reply:
Actually, I guess we could continue to support it for backward-compatibility if there is a check that the schema matches the inferred schema exactly and if there is a noticeable warning that this should be removed.

I would prefer to remove this support in v2. We already know that v2 will require a migration, so we may as well fix this while we can.

Reply:
How about this, let's support it in this migration PR. And we can discuss it in another followup PR, where test cases would be updated/added as well.

Reply:
Does this fail tests if it is removed?

Reply:
Yes, tests will fail.

Reply:
Okay, let's leave it in to get the majority of this work finished. I just filed SPARK-27960 for the ORC implementation as well, so I think that schema handling in the v2 file sources is a larger issue.